src/actions/exchangeActions.js

```diff
@@ -167,6 +167,19 @@ export const searchOffersAction = (fromAssetCode: string, toAssetCode: string, f
     );
     // we're requesting although it will start delivering when connection is established
     const { error } = await exchangeService.requestOffers(fromAssetCode, toAssetCode);
+
+    await api.fetchMoonPayOffers(fromAssetCode, toAssetCode, fromAmount).then((offer) => {
```

Comment:
Combining of .then() with await is not a good practise (only if you don't transform the data).
I would use the same approach as everywhere: `const moonPayOffers = await ...`

Reply:
The exception would be if you don't need to actually need to wait for the response in this flow, then consider removing `await`

Reply:
This caught my attention too, wouldn't mind even to make it truly async as it does not depend on the other results anyway. Would it be ok like that ?

Reply:
updated in 571513a